### PR TITLE
Proxy support with basic authentication has been added using system properties

### PR DESCRIPTION
**Pull request by jacombo**
Dokka is great plugin unfortunatelly can't be used in corporates behind proxy therefore proxy support with basic authentication has been added using system properties below

-Dhttp.proxyHost=host.domain.com
-Dhttp.proxyPort=8080
-Dhttps.proxyHost=host.domain.com
-Dhttps.proxyPort=8080
-DproxySet=true
-DproxyUser=user
-DproxyPassword=password

if proxySet is true and proxyUser with proxyPassword is set then basic proxy authentication is added

### CODE REVIEW

File: core/src/main/kotlin/Kotlin/ExternalDocumentationLinkResolver.kt [hunk 1]
```diff
@@ -46,6 +46,16 @@ class ExternalDocumentationLinkResolver @Inject constructor(
         connection.connectTimeout = timeout
         connection.readTimeout = timeout
 
+        // Add proxy basic authentication when proxyUser, proxyPassword is set and proxySet is true
+        val systemSettings = System.getProperties()
+        if (systemSettings.getValue("proxySet").toString() == "true"
```
Comment (semoro) [inline core/src/main/kotlin/Kotlin/ExternalDocumentationLinkResolver.kt]: As far, as I debugged, Gradle already support basic authentification, so no need for such code in Gradle runner plugin. 
Isn't it better to use `java.net.Authenticator`? It looks like more straightforward approach.

Comment (semoro) [reply core/src/main/kotlin/Kotlin/ExternalDocumentationLinkResolver.kt]: And why not to use `http.proxyUser` and so on?

**Comment by semoro** [comment]
Dokka already supports proxy authentication according to my comment in https://github.com/Kotlin/dokka/issues/213#issuecomment-393955468

File: core/src/main/kotlin/Kotlin/ExternalDocumentationLinkResolver.kt (hunk 1 above)
Comment (jacombo) [reply core/src/main/kotlin/Kotlin/ExternalDocumentationLinkResolver.kt]: could you debug and verify also  Maven + proxy + basic authentication ?

I will try also  will try use "systemProp.jdk.http.auth.tunneling.disabledSchemes" too and let you know

we are no using Graddle at the moment so we need to fix Maven first

**Comment by semoro** [comment]
Checked, doesn't work properly, looks like Dokka for Maven should use Maven transport API http://maven.apache.org/wagon/

**Comment by semoro** [comment]
So, anyway, changes is required

**Comment by kamildoleglo** [comment]
Dokka has been rewritten since the creation of this PR,  please open a new PR if the issue still persists ☺️

**Pull request closed without merging**